server/web/routes/proposals.js

```diff
@@ -322,6 +323,51 @@ async function getObjectFromS3(fileName) {
   });  
 }
 
+async function findAllProposalsInSameGroup(proposal, includeSelf = false) {
+  const history = [];
+  let original = null;
+  if (proposal.hasChild || proposal.parentId) {
+    // find all revised proposals
+    const revisedProposal = await Proposal.find({
+      groupId: proposal.groupId ? proposal.groupId : proposal._id.toString(),
+    });
+
+    if (proposal.groupId) {
+      // if current proposal is a revised proposal, add the original proposal to the list
+      original = await Proposal.findById(proposal.groupId);
+    } else {
+      // if current proposal is the original proposal, add itself to the list
+      original = proposal;
+    }
+
+    for (const prop of revisedProposal) {
+      if (!includeSelf && prop._id.toString() === proposal._id.toString()) continue;
+
+      history.push({
+        id: prop._id.toString(),
+        fileName: prop.fileName,
+        date: prop.createdAt,
+        dateString: prop.createdAt.toDateString(),
+      });
+    }
+
+    history.sort((a, b) => {
+      return b.date - a.date;
+    });
+  } else original = proposal;
+
+  return {
+    list: history,
+    hasHistory: history.length > Number(includeSelf) || proposal._id.toString() !== original._id.toString(),
```

Comment:
for hasHistory you can just check history.length > 0 since we never have to deal with the case when the proposal itself is included. Also get rid of the second condition, that's trivial and embedded in the logic of how we have initialized this history (list).